presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -416,16 +417,23 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
             groupingSetsSymbolsBuilder.add(groupingColumns.build());
         }
 
+        List<List<Symbol>> groupingSetsSymbols = groupingSetsSymbolsBuilder.build();
         // 2.c. Add a groupIdNode and groupIdSymbol if there are multiple grouping sets
         if (groupingSets.size() > 1) {
             Symbol groupIdSymbol = symbolAllocator.newSymbol("groupId", BIGINT);
-            GroupIdNode groupId = new GroupIdNode(idAllocator.getNextId(), subPlan.getRoot(), subPlan.getRoot().getOutputSymbols(), groupingSetsSymbolsBuilder.build(), groupIdSymbol);
+            GroupIdNode groupId = new GroupIdNode(idAllocator.getNextId(), subPlan.getRoot(), subPlan.getRoot().getOutputSymbols(), groupingSetsSymbols, groupIdSymbol);
             subPlan = new PlanBuilder(subPlan.getTranslations(), groupId, subPlan.getSampleWeight());
             distinctGroupingSymbolsBuilder.add(groupIdSymbol);
         }
         List<Symbol> distinctGroupingSymbols = distinctGroupingSymbolsBuilder.build().asList();
 
-        // 2.d. Mark distinct rows for each aggregate that has DISTINCT
+        // 2.d Determine the partitioning requirement for this aggregation (i.e. intersection of all grouping sets)
+        HashSet<Symbol> partitioningRequirement = new HashSet<>(groupingSetsSymbols.get(0));
```

Comment:
Do this in AddExchanges instead of recording it in the node. It's a property that only matters for the purpose of computing the distributed plan.